### PR TITLE
add ZZGTO4L_4f_NLO cards

### PR DESCRIPTION
**Pull request by Pupoin**
add cards for ZZGTo4L
MG process:
    generate p p > ell+ ell- ell+ ell- a [QCD]

Thanks!

### CODE REVIEW

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/ZZGTo4L_4f_NLO/ZZGTO4L_4f_NLO_run_card.dat [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+#***********************************************************************
+# Cuts on the charged leptons (e+, e-, mu+, mu-, tau+ and tau-)        *
+# (more specific gen cuts can be specified in SubProcesses/cuts.f)     *
+#***********************************************************************
+   3  = ptl     ! Min lepton transverse momentum
+  -1  = etal    ! Max lepton abs(pseudo-rap) (a value .lt.0 means no cut)
+   0  = drll    ! Min distance between opposite sign lepton pairs
+   0  = drll_sf ! Min distance between opp. sign same-flavor lepton pairs
+   1  = mll     ! Min inv. mass of all opposite sign lepton pairs
+   1  = mll_sf  ! Min inv. mass of all opp. sign same-flavor lepton pairs
```
Comment (sihyunjeon) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/ZZGTo4L_4f_NLO/ZZGTO4L_4f_NLO_run_card.dat:128]: You would get some divergence from gamma*->l+ l- with this 1 GeV mass cut. Is this what you really want for your analysis? If you are going to apply some mass cuts on mll near Z mass pole in RECO level, suggest to move it up to 50

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/ZZGTo4L_4f_NLO/ZZGTO4L_4f_NLO_run_card.dat [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+#***********************************************************************
+# Photon-isolation cuts, according to hep-ph/9801442                   *
+# When ptgmin=0, all the other parameters are ignored                  *
+#***********************************************************************
+  8  = ptgmin    ! Min photon transverse momentum
```
Comment (sihyunjeon) [inline bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/ZZGTo4L_4f_NLO/ZZGTO4L_4f_NLO_run_card.dat:133]: Your RECO level gamma pt cut will be ~ 8GeV?

Comment (Pupoin) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/ZZGTo4L_4f_NLO/ZZGTO4L_4f_NLO_run_card.dat:133]: yes, it seems that the ptgmin cut influence more to our analysis showing [here](https://indico.cern.ch/event/1013238/contributions/4252831/attachments/2219928/3759126/mmm_ptl10.png),  so we think it should more inclusive.

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/ZZGTo4L_4f_NLO/ZZGTO4L_4f_NLO_run_card.dat (hunk 1 above)
Comment (Pupoin) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/ZZGTo4L_4f_NLO/ZZGTO4L_4f_NLO_run_card.dat:128]: for z2, we have primary selection between 4~120 GeV, maybe we can chage mll_sf=4.

Comment (sihyunjeon) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/ZZGTo4L_4f_NLO/ZZGTO4L_4f_NLO_run_card.dat:128]: Then wouldn't it be better to use MadSpin by doing
generate p p > ell+ ell- z a
decay z > ell+ ell-
if you require mZ1 ~ 90 GeV?

To secure/claim more stats for your samples

Comment (sihyunjeon) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/ZZGTo4L_4f_NLO/ZZGTO4L_4f_NLO_run_card.dat:128]: And the computations would be faster

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/ZZGTo4L_4f_NLO/ZZGTO4L_4f_NLO_run_card.dat (hunk 2 above)
Comment (sihyunjeon) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/ZZGTo4L_4f_NLO/ZZGTO4L_4f_NLO_run_card.dat:133]: so the cut on your photons is > 10 GeV?

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/ZZGTo4L_4f_NLO/ZZGTO4L_4f_NLO_run_card.dat (hunk 1 above)
Comment (Pupoin) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/ZZGTo4L_4f_NLO/ZZGTO4L_4f_NLO_run_card.dat:128]: follow the advice from Olivier [here](https://answers.launchpad.net/mg5amcnlo/+question/696360), The symmetry factor can not be set in a consistent way in process p p > z l l a.

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/ZZGTo4L_4f_NLO/ZZGTO4L_4f_NLO_run_card.dat (hunk 2 above)
Comment (Pupoin) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/ZZGTo4L_4f_NLO/ZZGTO4L_4f_NLO_run_card.dat:133]: Yes, primary

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/ZZGTo4L_4f_NLO/ZZGTO4L_4f_NLO_run_card.dat (hunk 1 above)
Comment (sihyunjeon) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/ZZGTo4L_4f_NLO/ZZGTO4L_4f_NLO_run_card.dat]: Ah never knew this thanks for the clarification

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/ZZGTo4L_4f_NLO/ZZGTO4L_4f_NLO_run_card.dat (hunk 2 above)
Comment (sihyunjeon) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/ZZGTo4L_4f_NLO/ZZGTO4L_4f_NLO_run_card.dat]: Sorry I meant it as a RECO level photon pT cut.
As long as the GEN level cuts are safely inclusive (+ not really wasting a lot of phase space : cases like if you give mll>70 GeV in RECO level but give >10 GeV cut in GEN level which would yield a huge waste) the cut values are okay.

**Comment by sihyunjeon** [comment]
Just one minor question,
Is there a reason to enhance the bwcutoff to 30? The default should be 15 so I am just curious.

File: bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/ZZGTo4L_4f_NLO/ZZGTO4L_4f_NLO_run_card.dat (hunk 2 above)
Comment (Pupoin) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/ZZGTo4L_4f_NLO/ZZGTO4L_4f_NLO_run_card.dat]: > Just one minor question,
> Is there a reason to enhance the bwcutoff to 30? The default should be 15 so I am just curious.

Ops, I forget, there is no z, I change back to 15.

Comment (Pupoin) [reply bin/MadGraph5_aMCatNLO/cards/production/2017/13TeV/ZZGTo4L_4f_NLO/ZZGTO4L_4f_NLO_run_card.dat]: and, any other suggestion about runcard?

**Comment by sihyunjeon** [comment]
@agrohsje I think the cards are fine, one thing I would recommend in the future to the requestors would be use of the filters on mll. mll > 4 GeV for the analysis doesn't seem so efficient if one tags one of the Zs with onshell Z boson mass.

**Pull request merged**